src/providers/connectionStatus.tsx

```diff
@@ -1,23 +1,37 @@
-import React, { useCallback, useContext, useEffect, useRef, useState } from 'react'
+import React, { useCallback, useEffect, useRef } from 'react'
+import create from 'zustand'
 
 import { useSnackbar } from '@/providers/snackbars'
 
 export type ConnectionStatus = 'connected' | 'disconnected' | 'connecting'
 
-const ConnectionStatusContext = React.createContext<undefined | ConnectionStatusValue>(undefined)
-ConnectionStatusContext.displayName = 'ConnectionStatusContext'
-
 type ConnectionStatusValue = {
-  setNodeConnection: (connection: ConnectionStatus) => void
   nodeConnectionStatus: ConnectionStatus
   isUserConnectedToInternet: boolean
+  setNodeConnection: (connection: ConnectionStatus) => void
+  setInternetConnection: (isConnected: boolean) => void
 }
 
 const SNACKBAR_TIMEOUT = 15000
 
-export const ConnectionStatusProvider: React.FC = ({ children }) => {
-  const [nodeConnectionStatus, setNodeConnection] = useState<ConnectionStatus>('connecting')
-  const [isUserConnectedToInternet, setIsUserConnectedToInternet] = useState(true)
+export const useConnectionStatusStore = create<ConnectionStatusValue>((set) => ({
```

Comment:
This is using the default `zustand` `create`, not from out store utilities. We should use the common helper. This one also doesn't have `immer` so my guess is it's not fully working